sys/include/net/loramac.h

```diff
@@ -114,233 +98,428 @@ extern "C" {
+#define CONFIG_LORAMAC_DEFAULT_RX2_DR           (LORAMAC_DR_2)
 #endif
 #endif
 
 /**
- * @brief   Default second RX window frequency (in Hz)
+ * @brief   Default second RX (RX2) window frequency (in Hz)
  */
-#ifndef LORAMAC_DEFAULT_RX2_FREQ
+#ifndef CONFIG_LORAMAC_DEFAULT_RX2_FREQ
 #if IS_ACTIVE(CONFIG_LORAMAC_REGION_EU_868)
-#define LORAMAC_DEFAULT_RX2_FREQ               (869525000UL)
+#define CONFIG_LORAMAC_DEFAULT_RX2_FREQ         (869525000UL)
 #elif IS_ACTIVE(CONFIG_LORAMAC_REGION_IN_865)
-#define LORAMAC_DEFAULT_RX2_FREQ               (866550000UL)
+#define CONFIG_LORAMAC_DEFAULT_RX2_FREQ         (866550000UL)
 #endif
 #endif
 
 /**
  * @brief   Default LoRaMAC join procedure
+ *
+ * There are two options, Over The Air Activation (OTAA) results in device
+ * sending join request to the network whereas in the case of Activation By
+ * Personalization (ABP) the user enters the activation keys manually. OTAA is
+ * the preferred and most secure way to connect to a LoRaWAN network. For quick
+ * testing ABP is preferred as the device can transmit right away without
+ * waiting for the network to provision the keys.
  */
-#ifndef LORAMAC_DEFAULT_JOIN_PROCEDURE
-#define LORAMAC_DEFAULT_JOIN_PROCEDURE         (LORAMAC_JOIN_OTAA)
+#if IS_ACTIVE(CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE_OTAA)
+#define CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE   (LORAMAC_JOIN_OTAA)
+#elif IS_ACTIVE(CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE_ABP)
+#define CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE   (LORAMAC_JOIN_ABP)
+#endif
+
+#ifndef CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE
+#define CONFIG_LORAMAC_DEFAULT_JOIN_PROCEDURE   (LORAMAC_JOIN_OTAA)
 #endif
 
 /**
  * @brief   Default LoRaMAC join accept delay 1 (in seconds)
+ *
+ * Maximum wait time in end node to receive the `join-accept` message sent by
+ * the network in the first receive window. This is similar to the operation of
+ * RX1 window.
  */
-#ifndef LORAMAC_DEFAULT_JOIN_DELAY1
-#define LORAMAC_DEFAULT_JOIN_DELAY1            (5U)
+#ifndef CONFIG_LORAMAC_DEFAULT_JOIN_DELAY1
+#define CONFIG_LORAMAC_DEFAULT_JOIN_DELAY1      (5U)
 #endif
 
 /**
  * @brief   Default LoRaMAC join accept delay 2
+ *
+ * Maximum wait time in end node to receive the `join-accept` message sent by
+ * the network in the second receive window. This is similar to the operation of
+ * RX2 window.
  */
-#ifndef LORAMAC_DEFAULT_JOIN_DELAY2
-#define LORAMAC_DEFAULT_JOIN_DELAY2            (6U)
+#ifndef CONFIG_LORAMAC_DEFAULT_JOIN_DELAY2
+#define CONFIG_LORAMAC_DEFAULT_JOIN_DELAY2      (6U)
 #endif
 
 /**
  * @brief   Default max FCNT gap
+ *
+ * Frame counter deviation is the difference between the frame counter values
+ * transmitted by the End Node and the value stored in Network Server (NS).
  */
-#ifndef LORAMAC_DEFAULT_MAX_FCNT_GAP
-#define LORAMAC_DEFAULT_MAX_FCNT_GAP           (16384U)
-#endif
-
-/**
- * @brief   Default adaptive datarate ACK limit (in s)
- */
-#ifndef LORAMAC_DEFAULT_ADR_ACK_LIMIT
```

Comment:
These options are not used atm, but they could be easily added to the semtech pkg (and in the mid term to GNRC LoRaWAN). Maybe we could leave them there and simply not add them to Kconfig?

We can add a note to indicate they are not being used yet

Reply:
Yes, in fact, that was the initial plan. @leandrolanzieri  Hope that's okay.

Reply:
As long as there is a note saying that the options are not supported yet, there should be no issue I guess